File: project-03-realtime-dashboard/code-java/src/main/java/com/example/demo/service/DashboardService.java
package com.example.demo.service;

import java.time.LocalDate;
import java.util.List;

import org.springframework.stereotype.Service;

import com.example.demo.dto.DashboardResponseDto;
import com.example.demo.dto.ProductSalesRankDto;
import com.example.demo.repository.DashboardRepository;

@Service
public class DashboardService {
    private final DashboardRepository dashboardRepository;

    public DashboardService(DashboardRepository dashboardRepository) {
        this.dashboardRepository = dashboardRepository;
    }

    public DashboardResponseDto getDailyDashboard(LocalDate targetDate) {
        List<ProductSalesRankDto> ranks = dashboardRepository.getDashboardData(targetDate);
        return new DashboardResponseDto(targetDate, ranks);
    }
}

File: project-03-realtime-dashboard/code-kotlin/src/main/kotlin/com/example/demo/service/DashboardService.kt
package com.example.demo.service

import com.example.demo.dto.DashboardResponseDto
import com.example.demo.repository.DashboardRepository
import org.springframework.stereotype.Service
import java.time.LocalDate

@Service
class DashboardService(
    private val dashboardRepository: DashboardRepository
) {
    fun getDailyDashboard(targetDate: LocalDate): DashboardResponseDto {
        val ranks = dashboardRepository.getDashboardData(targetDate)
        return DashboardResponseDto(targetDate, ranks)
    }
}
